src/sss_client/pam_sss.c

```diff
@@ -1776,6 +1790,121 @@ static int prompt_oauth2(pam_handle_t *pamh, struct pam_items *pi)
     return PAM_SUCCESS;
 }
 
+static int prompt_passkey(pam_handle_t *pamh, struct pam_items *pi,
+                          const char *prompt_interactive, const char *prompt_touch)
+{
+    int ret;
+    const struct pam_conv *conv;
+    const struct pam_message *mesg[3] = { NULL, NULL, NULL };
+    struct pam_message m[3] = { {0}, {0}, {0} };
+    struct pam_response *resp = NULL;
+    int pin_idx = 0;
+    int msg_idx = 0;
+
+    ret = pam_get_item(pamh, PAM_CONV, (const void **) &conv);
+    if (ret != PAM_SUCCESS) {
+        return ret;
+    }
+    if (conv == NULL || conv->conv == NULL) {
+        logger(pamh, LOG_ERR, "No conversation function");
+        return PAM_SYSTEM_ERR;
+    }
+
+    pi->pam_authtok_type = SSS_AUTHTOK_TYPE_OAUTH2;
+	/* Interactive, prompt a message and wait before continuing */
+    if ((strcmp(prompt_interactive, "")) != 0) {
```

Comment:
Not sure if `(prompt_interactive != NULL && prompt_interactive[0] != '\0')` would be better?

Reply:
Fixed.